iam.tf

```diff
@@ -0,0 +1,59 @@
+resource "aws_iam_instance_profile" "this" {
+  name = "${var.environment}-openvpn"
+  role = aws_iam_role.this.name
+}
+
+resource "aws_iam_role" "this" {
+  name = "${var.environment}-openvpn"
+    assume_role_policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_iam_role_policy" "this" {
+  name   = "${var.environment}-openvpn"
+  role   = aws_iam_role.this.id
+  policy = <<-EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ssm:DescribeParameters",
+                "ssm:GetParameterHistory",
+                "ssm:DescribeDocumentParameters",
+                "ssm:GetParametersByPath",
+                "ssm:GetParameters",
+                "ssm:GetParameter",
+                "secretsmanager:ListSecrets",
+                "secretsmanager:GetSecretValue",
+                "ec2:AssociateAddress",
+                "ec2:DisassociateAddress",
+                "secretsmanager:DescribeSecret" 
```

Comment:
What is these permission needed for?

Reply:
The secret* permissions were removed. They are not needed. We need the ec2 permissions because the VPN server needs to keep the public IP so when the server starts, the first thing it does is to disassociate the elastic IP to the previous server (if there was one) and associates to itself.